Library/Homebrew/utils/curl.rb

```diff
@@ -139,15 +139,15 @@ def curl_output(*args, **options)
     # Check if a URL is protected by CloudFlare (e.g. badlion.net and jaxx.io).
     def url_protected_by_cloudflare?(details)
       [403, 503].include?(details[:status].to_i) &&
-        details[:headers].match?(/^Set-Cookie: __cfduid=/i) &&
-        details[:headers].match?(/^Server: cloudflare/i)
+        details[:headers]&.match?(/^Set-Cookie: __cfduid=/i) &&
+        details[:headers]&.match?(/^Server: cloudflare/i)
     end
 
     # Check if a URL is protected by Incapsula (e.g. corsair.com).
     def url_protected_by_incapsula?(details)
       details[:status].to_i == 403 &&
-        details[:headers].match?(/^Set-Cookie: visid_incap_/i) &&
-        details[:headers].match?(/^Set-Cookie: incap_ses_/i)
+        details[:headers]&.match?(/^Set-Cookie: visid_incap_/i) &&
```

Comment:
Why does this change result in sometimes-`nil` headers? I'd suggest an early return if `details[:headers]` is `nil` rather than doing a `&` each time.